dipy/workflows/io.py

```diff
@@ -97,3 +98,95 @@ def run(self, input_files,
+            return importlib.reload(sys.modules[module_path])
+        else:
+            return importlib.import_module(module_path)
+
+    def run(self, data_names, out_dir=''):
+        """Download files to folder and check their md5 checksums.
+
+        Parameters
+        ----------
+        data_names : variable string
+            Any number of Nifti1, bvals or bvecs files.
+        out_dir : string, optional
+            Output directory. Default: dipy home folder (~/.dipy)
+
+        """
+        if out_dir:
+            dipy_home = os.environ.get('DIPY_HOME', None)
+            os.environ['DIPY_HOME'] = out_dir
+
+        fetcher_module = self.load_module('dipy.data.fetcher')
+
+        available_data = dict([(name.replace('fetch_', ''), func)
+                               for name, func in getmembers(fetcher_module,
+                                                            isfunction)
+                               if name.lower().startswith("fetch_")
+                               if not len(getfullargspec(func).args)])
+
+        data_names = [name.lower() for name in data_names]
+
+        if 'all' in data_names:
+            for name, fetcher_function in available_data.items():
+                logging.info('------------------------------------------')
+                logging.info('Fetching at {0}'.format(name))
+                logging.info('------------------------------------------')
+                fetcher_function()
+
+        elif 'list' in data_names:
+            logging.info('Please, select between the following data names:'
+                         ' {0}'.format(', '.join(available_data.keys())))
+
+        else:
+            skipped_names = []
+            for data_name in data_names:
+                if data_name not in available_data.keys():
+                    skipped_names.append(data_name)
+                    continue
+
+                logging.info('------------------------------------------')
+                logging.info('Fetching at {0}'.format(data_name))
+                logging.info('------------------------------------------')
+                available_data[data_name]()
+
+            nb_success = len(data_names) - len(skipped_names)
+            print('\n')
+            logging.info('Fetched {0} / {1} Files '.format(nb_success,
+                                                           len(data_names)))
+            if skipped_names:
+                logging.warn('Skipped data name(s):'
+                             ' {0}'.format(' '.join(skipped_names)))
+                logging.warn('Please, select between the following data'
+                             ' names: {0}'.format(
+                                 ', '.join(available_data.keys())))
+
+        if out_dir:
+            if dipy_home:
+                os.environ['DIPY_HOME'] = dipy_home
+            else:
+                os.environ.pop('DIPY_HOME', None)
+
+            self.load_module('dipy.data.fetcher')
```

Comment:
```suggestion
            # We load the module again so that if we run another one of these in 
            # the same process, we don't have the env variable pointing to the 
            # wrong place
            self.load_module('dipy.data.fetcher')
```